src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -434,9 +450,6 @@ public static partial class MarshalHelpers
                 {
                     case NativeTypeKind.Array:
                         {
-                            if (isField || isReturn)
```

Comment:
removal of this check seems to have caused a regression compiling a test containing VT arrays to itself: https://github.com/dotnet/runtime/blob/9c3a78d835a5f1f7a32824becff9f56236806de5/src/tests/JIT/Methodical/VT/callconv/jumper.cs#L11. 

This is now failing outerloop jobs since @trylek switched them to crossgen2 today.

Reply:
Could you please share link to the log?

Reply:
Here is a [log ](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-53755-merge-4e4e6eea6bb047d19c/JIT.Methodical/console.bc1b4aa0.log?sv=2019-07-07&se=2021-06-25T03%3A43%3A03Z&sr=c&sp=rl&sig=XFHI2AE7lEtUOhQ0RIUvFGKvR0w8OXUo7lMVW7CgGdg%3D)of one of today's outerloop runs. Its generating a stackoverflow with this stack: 

```
         at Internal.TypeSystem.Interop.MarshalUtils.IsBlittableType(Internal.TypeSystem.TypeDesc)
         at Internal.TypeSystem.Interop.MarshalHelpers.GetArrayElementMarshallerKind(Internal.TypeSystem.ArrayType, Internal.TypeSystem.MarshalAsDescriptor, Boolean)
         at Internal.TypeSystem.Interop.MarshalHelpers.GetMarshallerKind(Internal.TypeSystem.TypeDesc, System.Nullable`1<Int32>, Internal.TypeSystem.EmbeddedSignatureData[], Internal.TypeSystem.MarshalAsDescriptor, Boolean, Boolean, Internal.TypeSystem.Interop.MarshallerType, Internal.TypeSystem.Interop.MarshallerKind ByRef)
         at Internal.TypeSystem.Interop.MarshalUtils.IsBlittableType(Internal.TypeSystem.TypeDesc)
         at Internal.TypeSystem.Interop.MarshalHelpers.GetArrayElementMarshallerKind(Internal.TypeSystem.ArrayType, Internal.TypeSystem.MarshalAsDescriptor, Boolean)
         at Internal.TypeSystem.Interop.MarshalHelpers.GetMarshallerKind(Internal.TypeSystem.TypeDesc, System.Nullable`1<Int32>, Internal.TypeSystem.EmbeddedSignatureData[], Internal.TypeSystem.MarshalAsDescriptor, Boolean, Boolean, Internal.TypeSystem.Interop.MarshallerType, Internal.TypeSystem.Interop.MarshallerKind ByRef)
         at Internal.TypeSystem.Interop.MarshalUtils.IsBlittableType(Internal.TypeSystem.TypeDesc)
```